ctp2_code/gfx/spritesys/director.cpp

```diff
@@ -447,7 +447,7 @@ class DQActionEffect : public DQAction {
 
 	// Lock behaviour
 	virtual void Unlock() {}
-	virtual bool IsUnlocked() { return true; }
+	virtual bool IsUnlocked() { return IsAnimationFinished(); }
```

Comment:
@MartinGuehmann do you think the delay in your AI test originates from this change?